scarlet/wavelet.py

```diff
@@ -16,304 +11,369 @@ class Starlet(object):
         images that have the same shape.
     """
 
-    def __init__(self, image=None, lvl=None, coefficients=None, direct=True):
+    def __init__(self, image, coefficients, generation, convolve2D):
         """ Initialise the Starlet object
 
-        Paramters
+        Parameters
+        ----------
         image: numpy ndarray
-            image to transform
-        lvl: int
-            number of starlet levels to use in the decomposition
-        starlet: array
-            Starlet transform of an array
-        direct: bool
-            if set to True, uses direct wavelet transform with the a trou algorithm.
-            if set to False, the transform is performed by convolving the image by the wavelet transform of a dirac.
+            Image in real space.
+        coefficients: array
+            Starlet transform of the image.
+        generation: int
+            The generation of the starlet transform (either `1` or `2`).
+        convolve2D: array-like
+            The filter used to convolve the image and create the wavelets.
+            When `convolve2D` is `None` this uses a cubic bspline.
         """
-        self.seed = None
-        # Transform method
-        self._direct = direct
-        if coefficients is None:
-            if image is None:
-                raise InputError(
-                    "At least an image or a set of coefficients should be provided"
-                )
-            else:
-                # Original shape of the image
-                self._image_shape = image.shape
-                # Padding shape for the starlet transform
-                if lvl is None:
-                    self._lvl = get_starlet_shape(image.shape)
-                else:
-                    self._lvl = lvl
-                if len(image.shape) == 2:
-                    image = image[np.newaxis, :, :]
-
-        else:
-            if len(np.shape(coefficients)) == 3:
-                coefficients = coefficients[np.newaxis, :, :, :]
-            self._image_shape = [coefficients.shape[0], *coefficients.shape[-2:]]
-            self._lvl = coefficients.shape[1]
-            if image is not None:
-                raise InputError(
-                    "Ambiguous initialisation: \
-                    Starlet objects should be instanciated either with an image of a set of coefficients, not both"
-                )
-
         self._image = image
         self._coeffs = coefficients
-        self._starlet_shape = [self._lvl, *self._image_shape[-2:]]
-        if self.seed is None:
-            self.seed = mk_starlet(self._starlet_shape)
-        self._norm = np.sqrt(np.sum(self.seed ** 2, axis=(-2, -1)))
+        self._generation = generation
+        self._convolve2D = convolve2D
+        self._norm = None
+
+    @staticmethod
+    def fromImage(image, scales=None, generation=2, convolve2D=None):
```

Comment:
I was always a fan of CamelCase, but for consistency let's call this thing `from_image`

Reply:
Yes, I've just been writing a lot of LSST code lately and got into the habit of writing camel case again (which I prefer too, but is not standard python).